zerver/views/users.py

```diff
@@ -100,13 +100,22 @@ def update_user_backend(request, user_profile, email,
 
     return json_success()
 
-def avatar(request, email):
+def avatar(request, email_or_id):
     # type: (HttpRequest, str) -> HttpResponse
     try:
-        user_profile = get_user_profile_by_email(email)
+        int(email_or_id)
+    except ValueError:
+        get_user_func = get_user_profile_by_email
+    else:
+        get_user_func = get_user_profile_by_id
+
+    try:
+        user_profile = get_user_func(email_or_id)
         avatar_source = user_profile.avatar_source
     except UserProfile.DoesNotExist:
         avatar_source = 'G'
+
+    email = user_profile.email if avatar_source == 'U' else email_or_id
```

Comment:
This is a small style thing, but we don't use the inline `if / else` pattern in the Zulip codebase; can you instead put this code in the try/except blocks for the two cases?